tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,2211 @@
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-south-http-south"], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "installation of packages failed"
+
+
+@pytest.fixture
+def enable_tls():
+    def _enable_tls(foglamp_url, wait_time, auth):
+        """
+        :param foglamp_url:
+        :param auth: None or password or certificate
+        """
+
+        if auth == 'password':
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}),
+                         headers={"authorization": PASSWORD_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}), headers={"authorization": PASSWORD_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+        elif auth == 'certificate':
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}),
+                         headers={"authorization": CERT_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}), headers={"authorization": CERT_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+        else:
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}))
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}))
+            r = conn.getresponse()
+            assert 200 == r.status
+
+        # Wait for foglamp server to start
+        time.sleep(wait_time * 2)
+
+    return _enable_tls
+
+
+@pytest.fixture
+def generate_password_token(asset_name, foglamp_url):
+    conn = http.client.HTTPConnection(foglamp_url)
+    conn.request("POST", "/foglamp/login", json.dumps({"username": "user", "password": "foglamp"}))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    jdoc = json.loads(r)
+    assert "Logged in successfully" == jdoc['message']
+    assert not jdoc['admin']
+    global PASSWORD_TOKEN
+    PASSWORD_TOKEN = jdoc["token"]
+
+
+@pytest.fixture
+def generate_certificate_token(asset_name, foglamp_url):
```

Comment:
generate_certificate_based_auth_token_for_admin

Reply:
We need to test: admin cert, user cert and an invalid

Reply:
I have covered the tests for login with admin and user cert in class.